src/baldr/graphtile.cc

```diff
@@ -317,6 +317,11 @@ void GraphTile::Initialize(const GraphId& graphid) {
   admins_ = reinterpret_cast<Admin*>(ptr);
   ptr += header_->admincount() * sizeof(Admin);
 
+  if (header_->has_osmids()) {
```

Comment:
you mentioned this in the description but yeah i think this is a breaking change. take for example the scenario where you use this new code to make tiles and then you send them to a device that is running the old code. when it goes to load the new tiles and they have this extra data, then the pointer arithmetic here will set the edgebins pointer to where you are writing the osm ids. because of this you pretty much have to write this at the end of the tile

Reply:
Everything below this data is indexed using an offset. The offsets in the new tiles (created using new code) all include 'the jump' over where the OSM ids are written. If access to the data beyond where the OSM ids is only ever done using the offsets (in the old code), then I think this will still work and is hence not a breaking change.

Apologies if I'm not understanding, but happy to move the data to the end of the tile in any case (just means that we have to access it via 'an offset', which complicates things).

Reply:
but have a look at the next snippet of code. you can see that `edge_bins_` is set to `ptr` which is always the first byte after the end of the admins. now after your change it might be the first byte after the node ids. so i agree with what your saying except that it isnt the case with the edge bins. concretely, someone who is running a version of valhalla before your code change has the following code:


```c++
...

  // Set a pointer to the admininstrative information list
  admins_ = reinterpret_cast<Admin*>(ptr);
  ptr += header_->admincount() * sizeof(Admin);

  // Set a pointer to the edge bin list
  edge_bins_ = reinterpret_cast<GraphId*>(ptr);

...
```

now imagine we merge your pr and then we create new tile data with it and enable osm node ids. then we give that person with old code, the new data. their old code will point the edge bins at the osm node ids. that is a breaking change. i think maybe you read in the graphtile header about the edgebin offsets and got confused. the edge bins are essentially an array within an array. so the "offsets" in the header arent with respect to the graphtile data, they are with respect to this `edge_bins_` pointer, which would be screwed up if we keep the code as you have it here.

in order to remove this burden of thought i think its always just easier to put new stuff on the end but technically you are right, there is a way to stay compatible and not put it on the end of the tile. its just not at this particular spot, it'd need to be after the edge bins which are also not fixed length and so at that point why not just put it on the end :smile: i mean i agree with you, itd be nice to keep all the fixed length stuff together but sadly the fools that designed it did not foresee this little snag :wink:

Reply:
Agreed... I had not paid attention enough to notice enough that this was not something like

```c++
complex_restriction_forward_ = tile_ptr + header_->complex_restriction_forward_offset();
```

like the rest of the ptr arithmetic.

Just out of interest, do you think this would make it possible to put it after the edge bins?

I'll move it to the end of the tile.

Reply:
Done.